packages/integration/src/awsS3/config.ts

```diff
@@ -28,7 +28,9 @@ export type AwsS3IntegrationConfig = {
   /**
    * The host of the target that this matches on, e.g. "amazonaws.com"
    *
-   * Currently only "amazonaws.com" is supported.
+   * If validateHost is true, "amazonaws.com" host is enforced. To test with localstack or similar AWS S3 emulators,
+   * setting validateHost to false allows hosts like "localhost:4566". Set ssl to false to access emulated S3
+   * endpoint over http (vs https). In that case, S3 urls would look like "http://<bucket>.localhost:4566/<path>".
```

Comment:
To be honest this seems a little bit magic, is it an established pattern?

Would another possibility be to have ab explicit `bucket` config value that needs to be set if it's no a `awamzonaws.com` url? With that we could remove the need for the `validateHost` config too

Reply:
I am pretty new to backstage, so excuse my ignorance, but it appears to me that the URL readers are supposed work off a single URL, and with `AwsS3UrlReader`, it should possible to setup a single host that would support reading from any bucket.

Currently `AwsS3UrlReader` expects S3 URLs in _virtual-hosted style_ form: `https://<bucket>.s3.<region>.amazonaws.com/<key>` (see [official docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html#path-style-access)). So I thought it'd make sense to keep this convention for non-amazonaws.com hosts. Alternatively, we may consider switching to _path-style_ form: https://s3.<region>.amazonaws.com/<bucket>/<key>, that would make non-amazonaws.com URLs look like less magic. Thoughts?

Reply:
As an idea: replace `validateHost` with `style` config, taking either `virtual-hosted` or `path` values.

Reply:
@Rugvip @leons727 i just realized - this is some vaguely related prior art in the techdocs backend:

https://github.com/backstage/backstage/blob/master/plugins/techdocs-backend/config.d.ts#L123

This is a direct parameter to the s3 client

https://github.com/backstage/backstage/blob/731e9293bd25e6b449d738e59b14b56b631bf16e/packages/techdocs-common/src/stages/publish/awsS3.ts#L136

There may be something to be said about aligning on basically that config format?

Reply:
I like the idea of aligning the config to what is closer to actual S3 config. It'd still require `ssl` flag to allow http connections.

Reply:
@freben I've pushed a change that uses `endpoint` and `s3ForcePathStyle` instead of `validateHost`/`ssl` in the config. It also removes `host` from awsS3 config schema.